src/glue.c

```diff
@@ -0,0 +1,163 @@
+#include "verify.h"
+#include <node_api.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+napi_status HandleError(napi_env env, struct Error *err) {
+  napi_status status;
+  napi_value error;       // The js Error object
+  napi_value errors;      // The js list of error strings
+  napi_value errorString; // Used to build the list of strings
+
+  char *msg = NULL; // used to build strings
+
+  if (err == NULL) {
+    status =
+        napi_throw_error(env, NULL, "Unknown exception verifying document");
+    VF_ERROR("received a NULL Error struct pointer\n");
+    return status;
+  }
+
+  status = napi_create_array(env, &errors);
+  if (status != napi_ok) {
+    VF_ERROR("could not create js array\n");
+    return status;
+  }
+
+  int i = 0;
+  while (err != NULL) {
+    msg = VF_err_fmt(err);
+
+    if (msg == NULL) {
+      VF_ERROR("could not format Error struct %s\n", err->reason);
+      return napi_generic_failure;
+    }
+
+    status = napi_create_string_utf8(env, msg, NAPI_AUTO_LENGTH, &errorString);
+    if (status != napi_ok) {
+      VF_ERROR("could not create js string for openssl error: %s\n", msg);
+      return status;
+    }
+
+    free(msg);
+    msg = NULL;
+
+    status = napi_set_element(env, errors, i, errorString);
+    if (status != napi_ok) {
+      VF_ERROR("could not set error %s as index %d on Errors.errors array");
+      return status;
+    }
+
+    if (err == err->next) {
+      VF_ERROR("found simple-cycle in error linked list\n");
+      break;
+    }
```

Comment:
yes.  as dicussed earlier, this is a simple cycle checker.  It detects the easiest way to have a badly constructed linked list, for example one where each node was constructed using the same heap memory.  Since this is a simple check, it's worth having.